subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/TaskPropertyInfo.java

```diff
@@ -96,7 +97,7 @@ public TaskPropertyValue getValue(Object rootObject) {
 
         final Object value = DeprecationLogger.whileDisabled(new Factory<Object>() {
             public Object create() {
-                return JavaReflectionUtil.method(Object.class, method).invoke(bean);
+                return DeferredUtil.unpack(JavaReflectionUtil.method(Object.class, method).invoke(bean));
```

Comment:
Doesn't this cause us to eagerly evaluate the property at this point (I'm not sure if this is much different from what we do now)?

Reply:
You are right, this may be a bit too early to unpack.